src/arrayMethodReduce.js

```diff
@@ -1,12 +1,19 @@
 'use strict';
 
-/**
- * Implement method Reduce
- */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let startValue = initialValue;
+
+    if (initialValue === undefined && arguments.length < 2) {
+      startValue = (typeof this[0] === 'number') ? 0 : '';
+    }
```

Comment:
Давай створимо змінну, яка буде визначати, з якого індексу починати. Якщо initialValue немає, то починати реба з 1 індексу (бо ми будемо брати перший елемент, як акумулятор). Якщо є, то 0
```suggestion
    let startInde = 0;
    
    if (arguments.length < 2) {
      startValue = this[i];
      startIndex = 1;
    }
```
А потім використаємо цей startIndex в циклі
```
for (let i = startIndex; i < this.length; i++) {
```